content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry/index.md

```diff
@@ -0,0 +1,245 @@
+
+A common approach to centralize logs is to use unstructured logs, parse them with regular expressions, and add contextual attributes.
+
+However, regular expression parsing is problematic. While initially easy, it becomes complex and fragile when handling all log fields, line breaks in exceptions, and unexpected log format changes. Parsing errors are inevitable with this method.
+
+## Solution
+
+The OpenTelemetry Java Instrumentation agent now offers an easy solution to convert logs from frameworks like SLF4J/Logback or Log4j2 into OTel-compliant JSON logs on stdout with all resource and log attributes.
+
+This is a true turnkey solution:
+
+* No code or dependency changes, just a few configuration adjustments typical for production deployment.
+* No complex field mapping in the log collector, just use the [OTLP/JSON connector](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/connector/otlpjsonconnector) to ingest the payload.
+* Automatic correlation between logs, traces, and metrics.
+
+This blog post shows how to set up this solution step by step.
+
+* In the first part, we'll show how to configure the Java application to output logs in the OTLP/JSON format. 
+* In the second part, we'll show how to configure the OpenTelemetry Collector to ingest the logs.
+* Finally, we'll show a Kubernetes-specific setup to handle container logs.
+
+## Reference architecture
+
+The deployment architecture looks like the following:
+
+![OTLP/JSON Architecture](otlpjson-architecture.png)
+
+## Configure Java Application to Output OTLP/JSON Logs
+
+Steps
+
+* No code changes needed. Continue using your preferred logging library, including templated logs, mapped diagnostic context, and structured logging.
+
+```java
+Logger logger = org.slf4j.LoggerFactory.getLogger(MyClass.class);
+...
+MDC.put("customerId", cusomerId);
+
+logger.info("Order {} successfully placed", orderId);
+
+logger.atInfo().
+   .addKeyValue("orderId", orderId)
+   .addKeyValue("outcome", "success")
+   .log("placeOrder");
+```
+
+* Export the logs captured by the OTel Java instrumentation to stdout using the OTel JSON format (aka [OTLP/JSON](https://opentelemetry.io/docs/specs/otlp/#json-protobuf-encoding)).
+  Configuration parameters for [Logback](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/logback/logback-appender-1.0/javaagent) and [Log4j](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/log4j/log4j-appender-2.17/javaagent) are optional but recommended.
+
+```bash
+# Tested with opentelemetry-javaagent v2.10.0
+#
+# Details on -Dotel.logback-appender.* params on documentation page:
+# https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/logback/logback-appender-1.0/javaagent
+
+java -javaagent:/path/to/opentelemetry-javaagent.jar \
+     -Dotel.logs.exporter=experimental-otlp/stdout \
```

Comment:
@zeitlinger do we want to use `-Dotel.logs.exporter` or `export OTEL_LOGS_EXPORTER`? Practitioners may look for the later and be confused to not find it in the text. 
Maybe we can add a mention that `export OTEL_LOGS_EXPORTER="experimental-otlp/stdout"` can be used.

Reply:
yes, good idea